### PR TITLE
#165382909 Users should be able to get top rated articles

### PR DESCRIPTION
**Pull request by davidokonji**
#### How Has This Been Tested?
Screenshot 2019-04-17 at 11 35 32

#### Checklist:
- [x]  add test for endpoint
- [x]  add endpoint to get top rated articles
- [x]  add documentation for endpoint
#### Story (PT)
[Delivers #165382909 ](https://www.pivotaltracker.com/story/show/165382909)

### CODE REVIEW

File: server/controllers/article.controller.js [hunk 1]
```diff
@@ -204,5 +210,37 @@ class ArticleController {
...
+        model: User,
+        as: 'author',
+        attributes: {
+          exclude: ['istokenreset', 'password', 'social', 'roleId', 'subscribed', 'id']
+        }
+      }]
+    });
+    const rate = allarticles.map(async (article) => {
+      const ratings = await getArticleRating(article.dataValues.id);
+      const rateObj = { article, ratings };
```
Comment (geekdanitek) [inline server/controllers/article.controller.js]: The ratings value should a property in the same object of the article

**Review by geekdanitek — CHANGES_REQUESTED** [review]

File: server/controllers/article.controller.js (hunk 1 above)
Comment (davidokonji) [reply server/controllers/article.controller.js]: @geekdanitek Thanks would fix that

**Review by geekdanitek — APPROVED** [review]

File: server/controllers/article.controller.js [hunk 2]
```diff
@@ -204,5 +210,52 @@ class ArticleController {
...
+        createdAt,
+        updatedAt,
+        author,
+        ratings
+      };
+      return rateObj;
+    });
+    const rateResponse = await Promise.all(rate);
+    const response = rateResponse.sort((a, b) => b.ratings - a.ratings);
+    return Response(res, 200, 'Retrived all top rated articles feed', { article: response });
```
Comment (Chiazokam) [inline server/controllers/article.controller.js]: Hi @davidokonji Great work. Just fix the typo at `Retri(e)ved`

**Review by chukwudidu — APPROVED** [review]
Good one.

File: server/controllers/article.controller.js (hunk 2 above)
Comment (davidokonji) [reply server/controllers/article.controller.js]: Thanks for the observation, fixed

File: server/controllers/article.controller.js [hunk 3]
```diff
@@ -300,5 +301,52 @@ class ArticleController {
...
+      },
+      include: [{
+        model: User,
+        as: 'author',
+        attributes: {
+          exclude: ['istokenreset', 'password', 'social', 'roleId', 'subscribed', 'id']
+        }
+      }]
+    });
+    const rate = allarticles.map(async (article) => {
```
Comment (samorahno) [inline server/controllers/article.controller.js:379]: Good job @davidokonji , kindly look into this variable naming and make it follow the right convention.

**Review by samorahno — CHANGES_REQUESTED** [review]

**Review by TheDrizzyWay — APPROVED** [review]
Good job.

**Review by samorahno — APPROVED** [review]

**Comment by BukkyOmo** [comment]
> Good job.

Great work

**Review by BukkyOmo — APPROVED** [review]

**Review by Chiazokam — APPROVED** [review]

**Review by paradisekelechi — APPROVED** [review]

**Pull request merged**